api/src/main/java/com/codeforcommunity/api/ITeamsProcessor.java

```diff
@@ -1,7 +1,16 @@
 package com.codeforcommunity.api;
 
 import com.codeforcommunity.auth.JWTData;
+import com.codeforcommunity.dto.team.*;
 
 public interface ITeamsProcessor {
   void disbandTeam(JWTData userData, int teamId);
+
+  void createTeam(JWTData userData, CreateTeamRequest request);
+
+  TeamDataResponse getTeam(JWTData userData, GetTeamRequest getTeamRequest);
+
+  void addGoal(JWTData userData, AddGoalRequest addGoalRequest);
+
+  void deleteGoal(JWTData userData, DeleteGoalRequest deleteGoalRequest);
```

Comment:
Delete goal doesn't have any JSON body and just has the goalId in the path. For these cases, creating a DTO doesn't save us any time because we don't use that `jsonBodyToDTO` method in `RestFunctions`. So instead of creating a new DTO just add a `Integer goalId` parameter to this interface instead